### ISSUE TITLE
SDL only includes directives when `.federation()` is enabled.

### ISSUE THREAD

**Issue by NfNitLoop** [async-graphql/async-graphql#1440]
## Expected Behavior

When I call [`Schema::sdl()`][1], I expect to see my custom [Type System Directive]s in its output.

[1]: https://docs.rs/async-graphql/latest/async_graphql/struct.Schema.html#method.sdl
[Type System Directive]: https://async-graphql.github.io/async-graphql/en/custom_directive.html#type-system-directives

## Actual Behavior

Directives are not included in the SDL output.

I played around with `sdl_with_option()` and found that enabling `.federation()` *does* output the directives:

```diff
> directive @testDirective(scope: String!, input: Int!, opt: Int) on FIELD_DEFINITION | OBJECT
```

… but that's not an acceptable workaround, because it also makes these unrelated changes.  (my API is not federated)

```diff
< schema {
<       query: Query
<       mutation: Mutation
< }
> extend schema @link(
>       url: "https://specs.apollo.dev/federation/v2.3",
>       import: ["@key", "@tag", "@shareable", "@inaccessible", "@override", "@external", "@provides", "@requires", "@composeDirective", "@interfaceObject"]
> )
> directive @include(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
> directive @skip(if: Boolean!) on FIELD | FRAGMENT_SPREAD | INLINE_FRAGMENT
```

## Steps to Reproduce the Problem

1. Copy/paste the example for a [Type System Directive] into your code.
2. Apply that directive to one of your field/object definitions as documented above.
3.  call `.sdl()` on your schema.  No directives in output.

## Specifications

- Version: 6.0.11
- Platform: arm64 / macOS
- Subsystem: ??? (I'm not sure what this is asking for.)

**Comment by sunli829** [comment]
fixed in #1407

**Comment by xamgore** [comment]
I believe, the issue can be closed now.
